backend/src/documents/upload_document.py

```diff
@@ -0,0 +1,46 @@
+import base64
+import os
+import uuid
+
+from src import context
+from src.storage import CloudStorage
+
+
+def upload_file_data(file_name, file_content, bucket_name, default_folder) -> str:
+    decoded_file_content = decode_file_content(file_content)
+    secure_filename, document_id = generate_secure_filename(file_name)
+    upload_file_to_cloud(decoded_file_content, secure_filename, file_name, bucket_name, default_folder)
+    return document_id
+
+
+def decode_file_content(file_content) -> bytes:
+    try:
+        decoded_file_data = base64.b64decode(file_content)
+    except Exception as e:
+        raise TypeError("Invalid file content encoding") from e
+    return decoded_file_data
+
+
+def generate_secure_filename(original_filename):
+    """
+    Generate a secure filename while preserving extension and adding uniqueness.
+    """
+    ext = os.path.splitext(original_filename)[1].lower() if "." in original_filename else ""
+
+    document_id = str(uuid.uuid4())
+
+    return f"{document_id}{ext}", document_id
+
+
```

Comment:
My problem with this now is the amount of parameters. I was thinking of using a model Class with getters/setters and maybe a builder to construct this instead. Let me know your thoughts on this @halprin.

Reply:
Sure, there's six arguments, but I'm not worried about it.  IMHO, one shouldn't put arguments into a holder unless there is a legitimate concept that holds all the arguments.

If you want, you could put the functionality of `upload_file_to_cloud` in the main `upload_file_data` function too.

Reply:
Moving the code into `upload_file_data` I think satisfies the issue for me. `upload_file_data` may have 1 more parameter but by default is isn't needed to be passed.